The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10585419416/job/29331977430) [(plain)](https://github.com/rust-lang/rust/commit/cea707d96033d5e14d3c2b287b0bedd2847db680/checks/29331977430/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 3.084 Building wheels for collected packages: reuse
#16 3.085   Building wheel for reuse (pyproject.toml): started
#16 3.329   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 3.330   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 3.331   Stored in directory: /tmp/pip-ephem-wheel-cache-4yyq1wq3/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 3.333 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.737 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.738 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.8s
---
   Compiling build_helper v0.1.0 (/checkout/src/tools/build_helper)
    Finished `dev` profile [unoptimized] target(s) in 27.29s
##[endgroup]
thread 'main' panicked at src/core/sanity.rs:387:10:
called `Result::unwrap()` on an `Err` value: "Cannot find any suitable upstream master branch"
Build completed unsuccessfully in 0:00:37
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 2/5:
##[group]Building bootstrap
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
thread 'main' panicked at src/core/sanity.rs:387:10:
called `Result::unwrap()` on an `Err` value: "Cannot find any suitable upstream master branch"
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 3/5:
##[group]Building bootstrap
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
thread 'main' panicked at src/core/sanity.rs:387:10:
called `Result::unwrap()` on an `Err` value: "Cannot find any suitable upstream master branch"
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 4/5:
##[group]Building bootstrap
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
thread 'main' panicked at src/core/sanity.rs:387:10:
called `Result::unwrap()` on an `Err` value: "Cannot find any suitable upstream master branch"
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:94: prepare] Error 1
Command failed. Attempt 5/5:
##[group]Building bootstrap
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
thread 'main' panicked at src/core/sanity.rs:387:10:
called `Result::unwrap()` on an `Err` value: "Cannot find any suitable upstream master branch"
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:94: prepare] Error 1
The command has failed after 5 attempts.
  local time: Tue Aug 27 20:37:40 UTC 2024
```

</details>

--- a/src/bootstrap/src/core/sanity.rs
+++ b/src/bootstrap/src/core/sanity.rs
@@ -382,7 +382,13 @@ $ pacman -R cmake && pacman -S mingw-w64-x86_64-cmake
         cmd_finder.must_have(s);
     }
 
-    warn_old_master_branch(&build.config.git_config(), &build.config.src)
-        .map_err(|e| e.to_string())
-        .unwrap();
+    // this warning is useless in CI,
+    // and CI probably won't have the right branches anyway.
+    if !build_helper::ci::CiEnv::is_ci() {
+        if let Err(e) = warn_old_master_branch(&build.config.git_config(), &build.config.src)
+            .map_err(|e| e.to_string())
+        {
+            eprintln!("unable to check if upstream branch is old: {e}");
+        }
+    }
 }

